test-framework/src/main/java/org/apache/brooklyn/test/framework/TestHttpCallImpl.java

```diff
@@ -77,13 +88,18 @@ public void start(Collection<? extends Location> locations) {
     }
 
     private void doRequestAndCheckAssertions(Map<String, Duration> flags, List<Map<String, Object>> assertions,
-                                             HttpAssertionTarget target, final String url) {
+                                             HttpAssertionTarget target, final HttpMethod method, final String url, final Map<String, String> headers, final String body) {
         switch (target) {
             case body:
                 Supplier<String> getBody = new Supplier<String>() {
                     @Override
                     public String get() {
-                        return HttpTool.getContent(url);
+                        try {
+                            return HttpTool.execAndConsume(HttpTool.httpClientBuilder().build(), createHttpMethod(method, url, headers, body)).getContentAsString();
```

Comment:
Minor: a couple of very long lines here. Probably worth splitting them up for readability (particularly as there is another method call at the end of this one, which can easily be missed if someone doesn't scroll).